### PR TITLE
Bug 1872971 - Move CODEOWNERS file to correct location

### PR DESCRIPTION
**Pull request by jonalmeida**
The CODEOWNERS file needs to be in [the root directory][0] for it to be evaluated. Since we've merged projects, the code and team structures have changed, so let's start off with a simple one that includes some known basics and iteratively add more directory paths from there.

⚠️ Currently, it's [not possible to add bots](https://github.com/orgs/community/discussions/23064#discussioncomment-3238917) to the file, so I removed this section below that would no longer apply. If we introduced a new account that is not registered as an official bot as we did with @MickeyMoz and the @mozilla-mobile/bots  group, then we could re-add it below if desired:

```
# Bot owned files
strings.xml                                 @MickeyMoz
Gecko.kt                                    @MickeyMoz
publicsuffixes                              @MickeyMoz
```

[0]: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-file-location

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1872971

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# You can also refer to a user by an email address that has been added
+# to their GitHub account, for example user@example.com.
+# https://help.github.com/articles/about-codeowners/
+
+# Release Engineering pipeline
+/.cron.yml                      @mozilla-mobile/releng
+/.taskcluster.yml               @mozilla-mobile/releng
+/android-components/automation/ @mozilla-mobile/releng @mozilla-mobile/aet @mozilla-mobile/QA
+/fenix/automation/              @mozilla-mobile/releng @mozilla-mobile/aet @mozilla-mobile/QA
+/focus-android/automation/      @mozilla-mobile/releng @mozilla-mobile/aet @mozilla-mobile/QA
```
Comment (jonalmeida) [inline CODEOWNERS:22]: @AaronMT @rpappalax I added the QA team to these directories based on what I saw in the activity git history, but I'm unsure if this the write team or if you would prefer another one.

Comment (jcristau) [reply CODEOWNERS:22]: I think releng can be removed from those `/*/automation/` directories, FWIW.

**Review by Amejia481 — APPROVED** [review]
✅

File: CODEOWNERS (hunk 1 above)
Comment (AaronMT) [reply CODEOWNERS]: Sorry I missed this, @mozilla-mobile/QA is fine for now. There's a bunch of @mozilla-mobile team maintenance we should look at cleaning up and adding people. We may adjust this to @mozilla-mobile/mobile-ci

**Pull request merged**